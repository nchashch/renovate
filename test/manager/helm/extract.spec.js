const { extractPackageFile } = require('../../../lib/manager/helm/extract');

describe('lib/manager/helm/extract', () => {
  describe('extractPackageFile()', () => {
    it('returns empty array', () => {
      const content = `
      dependencies:
        - name: redis
          version: 0.9.0
          repository: https://kubernetes-charts.storage.googleapis.com/
        - name: postgresql
          version: 0.8.1
          repository: https://kubernetes-charts.storage.googleapis.com/
      `;
      expect(extractPackageFile(content)).not.toBeNull();
      expect(extractPackageFile(content)).toMatchSnapshot();
    });
  });
});
