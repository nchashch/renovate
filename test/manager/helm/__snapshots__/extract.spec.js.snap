// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/helm/extract extractPackageFile() parses simple requirements.yaml correctly 1`] = `
Object {
  "chartName": "example",
  "datasource": "helm",
  "deps": Array [
    Object {
      "currentValue": "0.9.0",
      "depName": "redis",
      "helmRepository": "https://kubernetes-charts.storage.googleapis.com/",
    },
    Object {
      "currentValue": "0.8.1",
      "depName": "postgresql",
      "helmRepository": "https://kubernetes-charts.storage.googleapis.com/",
    },
  ],
}
`;
